examples/diesel/src/main.rs

```diff
@@ -43,46 +44,50 @@ struct RowsUpdated {
     rows: usize,
 }
 
-fn create_product_handler(mut state: State) -> Box<HandlerFuture> {
+fn create_product_handler(mut state: State) -> Pin<Box<HandlerFuture>> {
     let repo = Repo::borrow_from(&state).clone();
-    let f = extract_json::<NewProduct>(&mut state)
-        .and_then(move |product| {
-            repo.run(move |conn| {
-                // Insert the `NewProduct` in the DB
+    async move {
+        let product = match extract_json::<NewProduct>(&mut state).await {
+            Ok(product) => product,
+            Err(e) => return Err((state, e)),
+        };
+
+        let rows = match repo
```

Comment:
Can we move the `repo.run` out of the `match` and into a local var? It's a little hard to parse at a glance, considering this is an example.

Reply:
Sure.